### PR TITLE
feat: Added common use cases

### PR DESCRIPTION
**Pull request by Macbeth98**

### CODE REVIEW

File: docs/spl-docs/common-usecases.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
+I apologize for that. Here is the reordered list with only the 38 use cases you provided:
+
```
Comment (AndrewHess) [inline docs/spl-docs/common-usecases.md:2]: Delete this

File: docs/spl-docs/common-usecases.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+**Problem:** URLs are often encoded for transmission over the Internet, which can make them difficult to read and interpret when analyzing data. Encoded characters (e.g., `%3A` for `:`) can obscure the actual content of the URL.
+
+**Solution:** To make URLs readable and usable for analysis, encoded URLs can be decoded back to their original form using the [`urldecode`](evaluation-functions/text-functions.md/#urldecodeurl) function in Splunk. This process involves converting percent-encoded characters back to their corresponding characters. [View full Solution](evaluation-functions/text-functions.md/#use-case-example-4)
+
+### 30. Extracting Email Recipients from Logs
+
+**Problem:** In email transaction logs, recipient addresses are often stored in a single string, separated by semicolons. Analyzing
+
+ individual recipient behavior or response rates requires splitting these strings into separate values for each recipient.
```
Comment (AndrewHess) [inline docs/spl-docs/common-usecases.md:189]: Your sentence is split across paragraphs

File: docs/spl-docs/common-usecases.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
+I apologize for that. Here is the reordered list with only the 38 use cases you provided:
+
+# Common Use-Cases
+
+### 1. Identifying Potential Brute Force Attacks
+
+**Problem:** Detecting potential brute force attacks is crucial for maintaining network security. These attacks often involve repeated attempts to connect to critical services like SSH (port 22) or RDP (port 3389) from the same source IP, aiming to guess passwords and gain unauthorized access.
+
+**Solution:** To identify potential brute force attacks, a [`search`](search-command.md) command can be utilized to filter firewall logs for blocked connection attempts to SSH and RDP ports, count the attempts by source and destination IP, and highlight cases with a high number of attempts. [View full Solution](search-command.md/#use-case-example)
```
Comment (AndrewHess) [inline docs/spl-docs/common-usecases.md:9]: Does this link work for you?
```
[View full Solution](search-command.md/#use-case-example)
```
It doesn't work for me unless it's one of these:
```
[View full Solution](search-command/#use-case-example)
[View full Solution](search-command#use-case-example)
```

Even with those versions, it just goes to the page (not the section), but I think that's a separate frontend issue

File: docs/spl-docs/common-usecases.md [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+
+**Solution:** The [`top`](top-command.md) command can be used to find the most commonly purchased products for each customer segment, along with the count and percentage of total purchases. [View full Solution](top-command.md/#use-case-example)
+
+### 16. Analyzing Revenue from Expensive Products
+
+**Problem:** The goal is to identify and analyze expensive products (those with prices greater than $1000) to determine the total revenue, as well as the minimum, maximum, and average prices of these products across each product category.
+
+**Solution:** The solution involves using a combination of the [`where
+
+`](where-command.md) and `stats` commands in a Splunk search to filter and analyze the data. [View full Solution](where-command.md/#use-case-example)
```
Comment (AndrewHess) [inline docs/spl-docs/common-usecases.md:101]: Formatting is messed up 
image

File: docs/spl-docs/common-usecases.md (hunk 3 above)
Comment (Macbeth98) [reply docs/spl-docs/common-usecases.md:9]: This works for me `search-command#use-case-example`. It goes to the proper section.

Comment (AndrewHess) [reply docs/spl-docs/common-usecases.md:9]: Ok great, let's use links like that then. It must just be an issue on my end

**Review by AndrewHess — APPROVED** [review]

**Pull request merged**